src/format/parse.rs

```diff
@@ -532,6 +557,46 @@ fn test_parse() {
     check!(".  4",          [fix!(Nanosecond)]; INVALID);
     check!("  .4",          [fix!(Nanosecond)]; TOO_LONG); // no automatic trimming
 
+    // fixed: nanoseconds without the dot
+    check!("",             [fix!(Nanosecond3NoDot)]; ); // no field set, but not an error
+    check!("0",            [fix!(Nanosecond3NoDot)]; nanosecond: 0);
+    check!("4",            [fix!(Nanosecond3NoDot)]; nanosecond: 400_000_000);
+    check!("42",           [fix!(Nanosecond3NoDot)]; nanosecond: 420_000_000);
+    check!("421",          [fix!(Nanosecond3NoDot)]; nanosecond: 421_000_000);
+    check!("42195",        [fix!(Nanosecond3NoDot)]; nanosecond: 421_000_000); // ignores everything after 3 digits
+    check!("000000000547", [fix!(Nanosecond3NoDot)]; nanosecond: 0);
+    check!("4x",           [fix!(Nanosecond3NoDot)]; TOO_LONG);
+    check!("  4",          [fix!(Nanosecond3NoDot)]; INVALID);
+    check!(".421",         [fix!(Nanosecond3NoDot)]; INVALID);
+
+    check!("",             [fix!(Nanosecond6NoDot)]; ); // no field set, but not an error
+    check!("0",            [fix!(Nanosecond6NoDot)]; nanosecond: 0);
+    check!("4",            [fix!(Nanosecond6NoDot)]; nanosecond: 400_000_000);
+    check!("42",           [fix!(Nanosecond6NoDot)]; nanosecond: 420_000_000);
+    check!("421",          [fix!(Nanosecond6NoDot)]; nanosecond: 421_000_000);
+    check!("42195",        [fix!(Nanosecond6NoDot)]; nanosecond: 421_950_000);
+    check!("421950803",    [fix!(Nanosecond6NoDot)]; nanosecond: 421_950_000); // ignores everything after 6 digits
+    check!("000003547",    [fix!(Nanosecond6NoDot)]; nanosecond: 3000);
+    check!("000000547",    [fix!(Nanosecond6NoDot)]; nanosecond: 0);
+    check!("4x",           [fix!(Nanosecond6NoDot)]; TOO_LONG);
+    check!("  4",          [fix!(Nanosecond6NoDot)]; INVALID);
+    check!(".421",         [fix!(Nanosecond6NoDot)]; INVALID);
+
+    check!("",             [fix!(Nanosecond9NoDot)]; ); // no field set, but not an error
+    check!("0",            [fix!(Nanosecond9NoDot)]; nanosecond: 0);
+    check!("4",            [fix!(Nanosecond9NoDot)]; nanosecond: 400_000_000);
+    check!("42",           [fix!(Nanosecond9NoDot)]; nanosecond: 420_000_000);
+    check!("421",          [fix!(Nanosecond9NoDot)]; nanosecond: 421_000_000);
+    check!("42195",        [fix!(Nanosecond9NoDot)]; nanosecond: 421_950_000);
+    check!("421950803",    [fix!(Nanosecond9NoDot)]; nanosecond: 421_950_803);
+    check!("421950803547", [fix!(Nanosecond9NoDot)]; nanosecond: 421_950_803);
+    // check!("421950803547", [fix!(Nanosecond9NoDot), num!(Second)]; nanosecond: 421_950_803, second: 54); // don't skip digits that come after the 9
```

Comment:
did you want to uncomment this?

Reply:
Yeah, I'm fixing the test. I'll add another commit to the PR tomorrow